amlip_cpp/src/cpp/types/GenericType.cpp

```diff
@@ -0,0 +1,214 @@
+{
+    data_ = x.data_;
+    data_size_ = x.data_size_;
+
+    return *this;
+}
+
+GenericType& GenericType::operator =(
+        GenericType&& x)
+{
+    data_ = std::move(x.data_);
+    data_size_ = x.data_size_;
+
+    return *this;
+}
+
+bool GenericType::operator ==(
+        const GenericType& x) const
+{
+    return (data_ == x.data_ && data_size_ == x.data_size_);
+}
+
+bool GenericType::operator !=(
+        const GenericType& x) const
+{
+    return !(*this == x);
+}
+
+void* GenericType::data() const
+{
+    return data_;
+}
+
+uint32_t GenericType::data_size() const
+{
+    return data_size_;
+}
+
+const char* GenericType::type_name()
+{
+    return "GENERIC";
+}
+
+void GenericType::serialize(
+        eprosima::fastcdr::Cdr& scdr) const
+{
+    scdr << data_size_;
+    scdr.serializeArray(static_cast<uint8_t*>(data_), data_size_);
+}
+
+void GenericType::deserialize(
+        eprosima::fastcdr::Cdr& dcdr)
+{
+    // If data has been already allocated (it has been already deserialized), we free it
+    if (has_been_allocated_)
+    {
+        free(data_);
+    }
+
+    dcdr >> data_size_;
+
+    // Store enough space to deserialize the data
+    data_ = malloc(data_size_);
+    // Deserialize array
+    dcdr.deserializeArray(static_cast<uint8_t*>(data_), data_size_);
+
+    // Set as this data has been allocated by this class
+    has_been_allocated_.store(true);
+}
+
+void GenericType::serialize_key(
+        eprosima::fastcdr::Cdr&) const
+{
+}
+
+size_t GenericType::get_max_cdr_serialized_size(
+        size_t current_alignment)
+{
+    size_t initial_alignment = current_alignment;
+
+    current_alignment += 4 + eprosima::fastcdr::Cdr::alignment(current_alignment, 4);
+    current_alignment += 4 + eprosima::fastcdr::Cdr::alignment(current_alignment, 4);
+    // It needs an upper bound, but it will not be used
+    current_alignment += (100 * 1) + eprosima::fastcdr::Cdr::alignment(current_alignment, 1);
+
+    return current_alignment - initial_alignment;
+}
+
+size_t GenericType::get_cdr_serialized_size(
+        const GenericType& data,
+        size_t current_alignment)
+{
+    size_t initial_alignment = current_alignment;
+
+    current_alignment += 4 + eprosima::fastcdr::Cdr::alignment(current_alignment, 4);
+    current_alignment += 4 + eprosima::fastcdr::Cdr::alignment(current_alignment, 4);
+
+    if (data.data_size() > 0)
+    {
+        current_alignment += (data.data_size() * 1) + eprosima::fastcdr::Cdr::alignment(current_alignment, 1);
+    }
+
+    return current_alignment - initial_alignment;
+}
+
+size_t GenericType::get_key_max_cdr_serialized_size(
+        size_t current_alignment)
+{
+    return current_alignment;
+}
+
+bool GenericType::is_key_defined()
+{
+    return false;
+}
+
+bool GenericType::is_bounded()
+{
+    return false;
+}
+
+bool GenericType::is_plain()
+{
+    return false;
+}
+
+bool GenericType::construct_sample(
+        void* memory)
+{
+    return new (memory) GenericType();
```

Comment:
This should return `false` because type is not plain, so it should not be called.